cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/cube/DefaultCube.java

```diff
@@ -82,12 +85,14 @@ public void add(CubeFact fact) {
   @Override
   public void add(Collection<? extends CubeFact> facts) {
     List<Fact> toWrite = Lists.newArrayList();
+    int dimValuesCount = 0;
     for (CubeFact fact : facts) {
       for (Aggregation agg : aggregations.values()) {
         if (agg.accept(fact)) {
           List<DimensionValue> dimensionValues = Lists.newArrayList();
           for (String dimensionName : agg.getDimensionNames()) {
             dimensionValues.add(new DimensionValue(dimensionName, fact.getDimensionValues().get(dimensionName)));
+            dimValuesCount++;
```

Comment:
can do, dimValuesCount += dimensionValues.size() after this for loop, i am not sure if this metric will be helpful though.